src/frontends/pytorch/src/op/atan2.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/atan.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/greater_eq.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/subtract.hpp"
+#include "pt_framework_node.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_atan2_op(const NodeContext& context) {
+    num_inputs_check(context, 2, 2);
+    auto y = context.get_input(0);
+    auto x = context.get_input(1);
+
+    // handle the first condition : x>0
+    auto div_y_x = context.mark_node(make_shared<v1::Divide>(y, x));
+    auto atan = context.mark_node(make_shared<v0::Atan>(div_y_x));
+    auto const_zero = v0::Constant::create(element::i32, Shape{}, {0});
+    auto result = atan->output(0);
+
+    // handle the second condition : x<0 && y>=0
+    auto const_pi = v0::Constant::create(element::i32, Shape{}, {std::atan(1.0)*4});
+    auto is_x_negative = context.mark_node(make_shared<v1::Less>(x, const_zero));
```

Comment:
Less operator requires to have both inputs to have same type. This will fail if x would have type other than i32. Probably there would be some more issues in other ops